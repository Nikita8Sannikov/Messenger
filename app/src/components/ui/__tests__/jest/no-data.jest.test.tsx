import { render,screen } from "@testing-library/react"

import NoDataComponent from "@components/ui/no-data"

describe("No-data", () => {
    test("No-data shows correct", () => {
        render(<NoDataComponent text="Test"></NoDataComponent>)
        const dataElement = screen.getByTestId("no-data")
        expect(dataElement).toBeInTheDocument()
    })
    test("No-data has className", () => {
        render(<NoDataComponent text="Test"></NoDataComponent>)
        const dataElement = screen.getByTestId("no-data")
        expect(dataElement).toHaveClass("opacity-text no-items")
    })

    test("matches snapshot", () => {
        const { asFragment } = render(<NoDataComponent text="Test"></NoDataComponent>);

        // Снимаем снапшот компонента
        expect(asFragment()).toMatchSnapshot();
    });
})