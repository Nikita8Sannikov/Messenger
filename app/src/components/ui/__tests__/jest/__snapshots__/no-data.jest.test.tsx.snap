// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No-data matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="opacity-text no-items"
    data-testid="no-data"
  >
    Test
  </div>
</DocumentFragment>
`;
